PurchasesCoreSwift/Attribution/AttributionTypeFactory.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  AttributionTypeFactory.swift
+//  PurchasesCoreSwift
+//
+//  Created by Juanpe Catalán on 9/7/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+// TODO(Post-migration): switch this back to internal the class and all these protocols and properties.
+
+public typealias AttributionDetailsBlock = ([String: Any]?, Error?) -> Void
+
+@objc public enum FakeTrackingManagerAuthorizationStatus: Int {
+    case notDetermined = 0
+    case restricted
+    case denied
+    case authorized
+}
+
+class FakeAdClient: NSObject {
```

Comment:
Add a comment about why this class exists 📔  so future contributors aren't like "Lolwut?"

Reply:
Makes sense 🙂